imagebuildah/executor.go

```diff
@@ -352,6 +357,68 @@ func (b *Executor) resolveNameToImageRef(output string) (types.ImageReference, e
 	return imageRef, err
 }
 
+// buildCustomOutput takes imageID of the build and exports final built rootfs to specified directory or tar file.
+func (b *Executor) buildCustomOutput(ctx context.Context, imageID string, isDir bool, isStdout bool, path string) error {
+	didMount := false
+	image, err := internalUtil.LookupImage(b.systemContext, b.store, imageID)
+	if err != nil {
+		return errors.Wrapf(err, "failed while looking image %q", imageID)
+	}
+	defer func() {
+		if didMount {
+			// Log error as debug log here, since image could have been unmounted already
+			// or image is being used by another build, there is nothing we can do.
+			err := image.Unmount(false)
+			if err != nil {
+				logrus.Debugf("Unmount failed with: %v", err)
+			}
+		}
+	}()
+	mountPoint, err := image.Mountpoint()
+	if err != nil {
+		return errors.Wrapf(err, "failed getting mount point of image %q", imageID)
+	}
+	if mountPoint == "" {
+		mountPoint, err = image.Mount(ctx, nil, "")
+		if err != nil {
+			return errors.Wrapf(err, "failed mounting image %q", imageID)
+		}
+		didMount = true
+	}
+	input, err := archive.Tar(mountPoint, archive.Uncompressed)
+	if err != nil {
+		return errors.Wrapf(err, "failed while performing tar for %q", mountPoint)
+	}
+	defer input.Close()
+	if isDir {
+		if !filepath.IsAbs(path) {
+			cwd, err := os.Getwd()
+			if err != nil {
+				return err
+			}
+			path = filepath.Join(cwd, filepath.Clean(path))
+		}
```

Comment:
Wouldn’t it make sense to do this sometime early, during option processing or setup, so that we don’t do all the build work only for the saving to fail on a trivial path error?

Reply:
yes makes sense to me, fixed in latest commit.